appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AmountChooserPresenter.kt

```diff
@@ -29,6 +29,27 @@ class AmountChooserPresenter(private val appIconUrl: String, private val balance
         view?.sendKin(amount)
     }
 
+    override fun afterTextChanged(editable: Editable?) {
+        editable?.let {
+            if (editable.length >= 2) {
+                if (editable[0] == '0') {
+                    editable.delete(0, 1)
+                }
+            }
+            amount = 0
+            if (editable.isNotEmpty()) {
+                amount = try {
+                    editable.toString().toInt()
+                } catch (e: NumberFormatException) {
+                    editable.delete(editable.length - 2, editable.length - 1)
+                    editable.toString().toInt()
```

Comment:
can you not crash here again with NumberFormatException ?

Reply:
If you don't want to deal with try/catch you can also use: editable.toString().toIntOrNull() which will return null if the string is not an int

Reply:
no i cant crash again cause the perivous input was in integer limit
i will still need to trim the number to range on int
very very edge case